File: 2022/islands/islands.c
/*
 * This template is valid both in C and in C++,
 * so you can expand it with code from both languages.
 * NOTE: it is recommended to use this even if you don't
 * understand the following code.
 */

#include <stdio.h>
#include <assert.h>

// constraints
#define MAXN 1000

// input data
int R, C, i, j;
int M[MAXN][MAXN];

int main() {
//  uncomment the following lines if you want to read/write from files
//  freopen("input.txt", "r", stdin);
//  freopen("output.txt", "w", stdout);

    assert(2 == scanf("%d %d", &R, &C));
    for(i=0; i<R; i++)
        for (j=0; j<C; j++)
            assert(1 == scanf("%d", &M[i][j]));

    // insert your code here

    printf("%d\n", 42); // print the result
    return 0;
}

File: 2022/algobadge/islands/islands.pas
{ NOTE: it is recommended to use this even if you don't understand the following code. }

{ constraints }
const
    MAXN = 1000;

{ input data }
var
    R, C, i, j  : longint;
    M           : array[0..MAXN-1, 0..MAXN-1] of longint;

begin
{
    uncomment the following lines if you want to read/write from files
    assign(input,  'input.txt');  reset(input);
    assign(output, 'output.txt'); rewrite(output);
}

    readln(R, C);
    for i:=0 to R-1 do
    begin
        for j:=0 to C-1 do
            read(M[i,j]);
        readln();
    end;

    { insert your code here }

    writeln(42); { print result }
end.
